core/domain/platform_parameter_domain.py

```diff
@@ -35,3 +46,935 @@ class PlatformParameterChange(change_domain.BaseChange):
         'required_attribute_names': ['new_rules'],
         'optional_attribute_names': []
     }]
+
+
+class EvaluationContext(python_utils.OBJECT):
+    """Domain object representing the context for parameter evaluation."""
+
+    def __init__(
+            self, client_platform, client_type, browser_type,
+            app_version, user_locale, mode):
+        self._client_platform = client_platform
+        self._client_type = client_type
+        self._browser_type = browser_type
+        self._app_version = app_version
+        self._user_locale = user_locale
+        self._mode = mode
+
+    @property
+    def client_platform(self):
+        """Returns client platform.
+
+        Returns:
+            str. The client platform.
+        """
+        return self._client_platform
+
+    @property
+    def client_type(self):
+        """Returns client type.
+
+        Returns:
+            str. The client type.
+        """
+        return self._client_type
+
+    @property
+    def browser_type(self):
+        """Returns client browser type.
+
+        Returns:
+            str. The client browser type.
+        """
+        return self._browser_type
+
+    @property
+    def app_version(self):
+        """Returns client application version.
+
+        Returns:
+            str. The client application version.
+        """
+        return self._app_version
+
+    @property
+    def user_locale(self):
+        """Returns client locale.
+
+        Returns:
+            str. The client locale.
+        """
+        return self._user_locale
+
+    @property
+    def mode(self):
+        """Returns the mode Oppia is running on.
+
+        Returns:
+            str. The the mode Oppia is running on.
+        """
+        return self._mode
+
+    @classmethod
+    def create_from_dict(cls, client_context_dict, server_context_dict):
+        """Creates a new EvaluationContext object by combining both client side
+        and server side context.
+
+        Args:
+            client_context_dict: dict. The client side context.
+            server_context_dict: dict. The server side context.
+
+        Returns:
+            EvaluationContext. The corresponding EvaluationContext domain
+            object.
+        """
+        return cls(
+            client_context_dict.get('client_platform'),
+            client_context_dict.get('client_type'),
+            client_context_dict.get('browser_type'),
+            client_context_dict.get('app_version'),
+            client_context_dict.get('user_locale'),
+            server_context_dict.get('mode'),
+        )
+
+
+class PlatformParameterFilter(python_utils.OBJECT):
+    """Domain object for filters in platform parameters."""
+
+    SUPPORTED_FILTER_TYPE = [
+        'mode', 'user_locale', 'client_platform', 'client_type',
+        'browser_type', 'app_version',
+    ]
+    VERSION_EXPR_REGEXP = re.compile(r'^(>|<|=|>=|<=)?(\d+(?:\.\d+)*)$')
```

Comment:
Lets disallow not specifying the comparison, so 
```suggestion
    SUPPORTED_FILTER_TYPE = [
        'mode', 'user_locale', 'client_platform', 'client_type',
        'browser_type', 'app_version',
    ]
    VERSION_EXPR_REGEXP = re.compile(r'^(>|<|=|>=|<=)(\d+(?:\.\d+)*)$')
```

Reply:
Done.